readthedocs/vcs_support/backends/git.py

```diff
@@ -123,11 +123,18 @@ def clone(self):
 
     @property
     def tags(self):
+        versions = []
         repo = git.Repo(self.working_dir)
-        versions = [
-            VCSVersion(self, str(tag.commit), str(tag))
-            for tag in repo.tags
-        ]
+        for tag in repo.tags:
+            try:
```

Comment:
I think it should be rewritten as following
```python
repo = git.Repo(self.working_dir)
versions = []
for tag in repo.tags:
     tag_obj = tag.tag
     commit = tag_obj.object
     tag_name = tag_obj.tag
     versions.append(VCSVersion(self, str(commit), tag_name))
```

[`tag.TagObject`](https://gitpython.readthedocs.io/en/stable/reference.html#git.objects.tag.TagObject) has the `object` option which actually points to the commit. It will not raise `valueError` like the [`Refs.tag.commit`](https://gitpython.readthedocs.io/en/stable/reference.html#git.refs.tag.TagReference.commit)

Reply:
Yeah, I found that when I was researching about this but... is this something useful for Read the Docs? I mean, that blob object where the tag is pointing is something we can build?

I considered that and I thought that it will be useless/tricky/not valid for Read the Docs to consider that as a tag (version) that we want to build.

What do you think?

Reply:
I think as its a feature of git, we should support it. Many projects may use this feature and tag without commit is also in linux kernel. So from RTD point of view, its a tag and we should not consider whethere the tag is pointed to any commit or blob!

Reply:
A tag can point to a blob, a tree, or even another tag. However, the vast majority of tags across all repos point to commits. A repository that has a non-commit tag is using it for a special purpose (say, it points to a blob of the public key used by the project). With the admittedly limited examples of non-commit tags I've seen none of them could be built as documentation by Read the Docs.

I think it's safe to completely ignore non-commit tags. I would love to see examples of non-commit tags that would somehow be sensible to Read the Docs.

Reply:
> I think it's safe to completely ignore non-commit tags.

At least until somebody requests it. The fact that nobody has requested it and that non-commit tags caused a repo to fail building on Read the Docs is definitely something to consider as well.

Reply:
I believe if we can support the non commit tags, we should support it. We should not wait till someone open a issue with "My build is not working because my tag does not belong to a commit".

In the solution I provided, we can still support non commit tag as well as commit tag. So why do we exclude the non commit tag?

Reply:
It won't work. It will just fail later. When you attempt to checkout a non-commit in git like we do when we attempt to build documentation, you'll get the error:

    fatal: reference is not a tree: TAG-NAME

Reply:
Oh! I understand! Than I think its better to not have them in Version! Thanks @davidfischer